Client/Frontend/Browser/PageZoom/PageZoomSettingsView.swift

```diff
@@ -0,0 +1,197 @@
+              if preferredZoomLevel == zoomLevel {
+                Image(systemName: "checkmark")
+                  .font(.system(.footnote).weight(.medium))
+                  .foregroundColor(Color(UIColor.braveLabel))
+              }
+            }
+          })
+        }
+      }
+      .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      
+      if !domains.isEmpty {
+        Section(
+          header: Text(Strings.PageZoom.resetAllDescription)
+            .font(.footnote)
+            .foregroundColor(Color(UIColor.secondaryBraveLabel))
+        ) {
+          Button(Strings.PageZoom.resetAll) {
+            PageZoomSettingsView.allDomains(context).forEach({
+              $0.zoom_level = nil
+            })
+            
+            try? context.save()
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+            NotificationCenter.default.post(name: PageZoomView.notificationName, object: nil)
+          }
+        }
+        .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      }
+    }
+    .navigationTitle(Strings.PageZoom.settingsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .listStyle(.insetGrouped)
+  }
+  
+  func normalizeZoomLevels() {
+    domains.forEach({
+      if $0.zoom_level?.doubleValue == Preferences.General.defaultPageZoomLevel.value {
+        $0.zoom_level = nil
+      }
+    })
+    
+    try? context.save()
+    NotificationCenter.default.post(name: PageZoomView.notificationName, object: nil)
+  }
+  
+  private func resetDefaultZoomLevel(zoomLevel: Double) {
+    Preferences.General.defaultPageZoomLevel.value = zoomLevel
+    defaultZoomLevel = zoomLevel
+    
+    domains = PageZoomSettingsView.filteredDomains(context)
+  }
+  
+  private func resetItemZoomLevel(at offsets: IndexSet) {
+    offsets.forEach({
+      domains[$0].zoom_level = nil
+    })
+    
+    try? context.save()
+    
+    domains = PageZoomSettingsView.filteredDomains(context)
+    NotificationCenter.default.post(name: PageZoomView.notificationName, object: nil)
+  }
+  
+  private static func allDomains(_ context: NSManagedObjectContext) -> [Domain] {
+    let request = NSFetchRequest<Domain>().then {
+      $0.entity = Domain.entity(context)
+      $0.sortDescriptors = [NSSortDescriptor(key: "url", ascending: true)]
+    }
+    
+    do {
+      let domains = try context.fetch(request)
+      return domains.sorted(by: { $0.host < $1.host })
+    } catch {
+      return []
+    }
+  }
+  
+  private static func filteredDomains(_ context: NSManagedObjectContext) -> [Domain] {
+    let request = NSFetchRequest<Domain>().then {
+      $0.entity = Domain.entity(context)
+      $0.sortDescriptors = [NSSortDescriptor(key: "url", ascending: true)]
+      $0.predicate = NSPredicate(format: "zoom_level != nil")
+    }
+    
+    do {
+      let domains = try context.fetch(request)
+      return domains.sorted(by: { $0.host < $1.host })
```

Comment:
Why not add additional sort descriptor instead of sorting it here?

Reply:
Fixed.